harvester_e2e_tests/integration/test_vm_functions.py

```diff
@@ -596,6 +613,287 @@ def test_create_stopped_vm(api_client, stopped_vm, wait_timeout):
     assert 404 == code, (code, data)
 
 
+@pytest.mark.p0
+@pytest.mark.virtualmachines
+class TestVMResource:
+    @pytest.mark.parametrize("res_type", ["cpu", "memory"])
+    def test_create_schedule_on_maximum(
+        self, api_client, unique_vm_name, vm_checker, vm_calc, image,
+        unset_cpu_memory_overcommit, res_type
+    ):
+        # get the node having the maximum resource
+        code, data = api_client.hosts.get()
+        nodes_res = [(n['metadata']['name'], vm_calc.node_resources(n)['schedulable'])
+                     for n in data['data']]
+        nodes_res = sorted(nodes_res, key=lambda n: n[1][res_type], reverse=True)
+        expected_host, expected_res = next((cn, nd[res_type] + (cd[res_type] - nd[res_type]) / 2)
+                                           for (cn, cd), (nn, nd) in zip(nodes_res, nodes_res[1:]))
+
+        # Calculate the maximum resource
+        vm_spec, namespace = api_client.vms.Spec(1, 2), 'default'
+        vm_spec.add_image("disk-0", image['id'])
+        data = vm_spec.to_dict(unique_vm_name, namespace)
+
+        if res_type == 'cpu':
+            exp = -2
+            data['spec']['template']['spec']['domain'][res_type]['cores'] = int(expected_res) + 1
```

Comment:
when `cd["cpu"] == nd["cpu"]` then `expected_res == nd["cpu"] == cd["cpu"]`, thus `int(expected_res) + 1` will exceed `cd["cpu"]`. is that expected?

Reply:
The `expected_res` would be a `float`, and it would be used in **resources.limits** and **resources.requests**.
As the snapshot below, the `cores` would only expect an **Integer** and showing the number in OS (like `lscpu`), so that's why we do `int(expected_res) + 1`, we can limit/request VM's CPU be something like `11.3`, but OS would need show `12`.

![image](https://github.com/harvester/tests/assets/5169694/ee44da8d-af53-4309-856e-3355c0b56a84)